engines/sci/parser/vocabulary.cpp

```diff
@@ -188,9 +188,59 @@ bool Vocabulary::loadParserWords() {
 		seeker += 3;
 	}
 
+	// Russian translation can contain translated vocab in special format
+	if (g_sci->getLanguage() == Common::RU_RUS)
+		loadTranslatedWords();
+
 	return true;
 }
 
+void Vocabulary::loadTranslatedWords()
+{
+	// This is special fan made format similar to VOCAB.000 (see
+	// https://wiki.scummvm.org/index.php?title=SCI/Specifications/SCI_in_action/Parser#Vocabulary_file_formats)
+	// but all characters used are in the upper character range (80h..FFh)
+
+	Resource *resource = _resMan->findResource(ResourceId(kResourceTypeVocab, VOCAB_RESOURCE_SCUMM_LOC_VOCAB), 0);
+	if (!resource) return;
+	
+	char currentWord[VOCAB_MAX_WORDLENGTH] = "";
+	int currentWordPos = 0;
+
+	uint32 seeker = 0;
+	while (seeker < resource->size()) {
+		byte c;
+
+		currentWordPos = resource->getUint8At(seeker++); // Parts of previous words may be re-used
+
+		do {
+			if (seeker == resource->size()) {
```

Comment:
I would write here >=. You are reading `uint` which multibyte, thus, theoretically, you can go well beyond size.

Reply:
I reading uint8, this is one byte. Same check is in line 156.